jerry-core/lit/lit-char-helpers.c

```diff
@@ -98,6 +99,40 @@ search_char_in_interval_array (ecma_char_t c,               /**< code unit */
 } /* search_char_in_interval_array */
 
 /**
+ * Print a specified character.
+ */
+void
+lit_char_put (ecma_char_t c) /**< code unit */
+{
+  JERRY_STATIC_ASSERT (sizeof (c) == 2,
+                       size_of_code_point_must_be_equal_to_2_bytes);
+
+  if (c != LIT_CHAR_NULL && c <= LIT_UTF8_1_BYTE_CODE_POINT_MAX)
+  {
+    jerry_port_putchar ((char) c);
+  }
+  else
+  {
+    uint32_t byte_high = (uint32_t) JRT_EXTRACT_BIT_FIELD (ecma_char_t, c,
+                                                           JERRY_BITSINBYTE,
+                                                           JERRY_BITSINBYTE);
+    uint32_t byte_low = (uint32_t) JRT_EXTRACT_BIT_FIELD (ecma_char_t, c,
+                                                          0,
+                                                          JERRY_BITSINBYTE);
+
+    char out_buf[5];
+    sprintf(out_buf, "%02x%02x", byte_high, byte_low);
+
+    jerry_port_putchar ('\\');
+    jerry_port_putchar ('u');
+    jerry_port_putchar (out_buf[0]);
```

Comment:
I don't love this, but it's the simplest way to achieve what I am trying to achieve without adding more port APIs.